services/src/datasets/external/gbif.rs

```diff
@@ -197,6 +201,129 @@ impl GbifDataProvider {
             .collect::<Vec<_>>())
     }
 
+    async fn get_datasets_search_items(
+        &self,
+        limit: &u32,
+        offset: &u32,
+        path: &str,
+        search_string: &str,
+    ) -> Result<Vec<CollectionItem>> {
+        let taxonrank = path
+            .split_once('/')
+            .map_or_else(String::new, |(taxonrank, _)| taxonrank.to_string());
+        ensure!(
+            ["family", "genus", "species"].contains(&taxonrank.as_str()),
+            crate::error::InvalidPath
+        );
+        let path = path.split_once('/').map_or_else(|| "", |(_, path)| path);
+        let filters = GbifDataProvider::get_filters(path);
+        let conn = self.pool.get().await?;
+        let query = &format!(
+            r#"
+            SELECT {taxonrank}, COUNT(*)
+            FROM {schema}.occurrences
+            WHERE {taxonrank} IN
+                (
+                    SELECT DISTINCT canonicalname
+                    FROM {schema}.species
+                    WHERE taxonrank = $1{filter} AND canonicalname LIKE $4
+                )
+            GROUP BY {taxonrank}
+            ORDER BY {taxonrank} 
+            LIMIT $2
+            OFFSET $3;
+            "#,
+            schema = self.db_config.schema,
+            filter = filters
```

Comment:
Yes, only the valid column names can occur here since they're derived from the path depth. The filter values input by the user are passed as parameters to the prepared statement, not directly in the query string